### ISSUE TITLE
Vite: cannot resolve asset path in turborepo monorepo

### ISSUE THREAD

**Issue by lithdew** [remix-run/remix#7927]
### What version of Remix are you using?

2.2.0

### Are all your remix dependencies & dev-dependencies using the same version?

- [X] Yes

### Steps to Reproduce

1. `git clone https://github.com/lithdew/remix-vite-monorepo.git`
2. `npm install`
3. `npm run dev`

### Expected Behavior

It works.

### Actual Behavior

```console
% npm run dev

> dev
> vite dev


  VITE v4.5.0  ready in 597 ms

  ➜  Local:   http://localhost:5173/
  ➜  Network: use --host to expose
  ➜  press h to show help

  ⚠️  Remix support for Vite is unstable
     and not recommended for production

9:53:10 PM [vite] Internal server error: Cannot resolve asset path "/Users/dew/remix-vite-monorepo/remix-vite-monorepo/node_modules/@remix-run/dev/dist/config/defaults/entry.server.node.tsx" outside of root directory "/Users/dew/remix-vite-monorepo/remix-vite-monorepo/apps/web".
      at resolveFileUrl (/Users/dew/remix-vite-monorepo/remix-vite-monorepo/node_modules/@remix-run/dev/dist/vite/plugin.js:74:11)
      at getServerEntry (/Users/dew/remix-vite-monorepo/remix-vite-monorepo/node_modules/@remix-run/dev/dist/vite/plugin.js:197:51)
      at async Context.load (/Users/dew/remix-vite-monorepo/remix-vite-monorepo/node_modules/@remix-run/dev/dist/vite/plugin.js:434:20)
      at async Object.load (file:///Users/dew/remix-vite-monorepo/remix-vite-monorepo/node_modules/vite/dist/node/chunks/dep-bb8a8339.js:44323:32)
      at async loadAndTransform (file:///Users/dew/remix-vite-monorepo/remix-vite-monorepo/node_modules/vite/dist/node/chunks/dep-bb8a8339.js:54954:24)
      at async instantiateModule (file:///Users/dew/remix-vite-monorepo/remix-vite-monorepo/node_modules/vite/dist/node/chunks/dep-bb8a8339.js:55951:10)
```

**Comment by izznatsir** [comment]
Could you make a reproduction repo? I'm using `react-aria-components` with remix vite plugin without a problem.

**Comment by lithdew** [comment]
> Could you make a reproduction repo? I'm using `react-aria-components` with remix vite plugin without a problem.

Oh interesting, would you be able to provide the configs you have for your project?

I'll be able to make the reproduction repo in a bit.

**Comment by lithdew** [comment]
While making the reproduction repo, I think I found the source of the bug. I explicitly created the Remix Vite repo inside of a Turborepo mono repository. I believe what Vite was complaining about was that the "react" dependency was in the node_modules of the monorepo folder rather than the node_modules of the Remix project folder instead.

**Comment by lithdew** [comment]
Created a reproduction here: https://github.com/lithdew/remix-vite-monorepo

```console
% npm run dev

> dev
> vite dev


  VITE v4.5.0  ready in 597 ms

  ➜  Local:   http://localhost:5173/
  ➜  Network: use --host to expose
  ➜  press h to show help

  ⚠️  Remix support for Vite is unstable
     and not recommended for production

9:53:10 PM [vite] Internal server error: Cannot resolve asset path "/Users/dew/remix-vite-monorepo/remix-vite-monorepo/node_modules/@remix-run/dev/dist/config/defaults/entry.server.node.tsx" outside of root directory "/Users/dew/remix-vite-monorepo/remix-vite-monorepo/apps/web".
      at resolveFileUrl (/Users/dew/remix-vite-monorepo/remix-vite-monorepo/node_modules/@remix-run/dev/dist/vite/plugin.js:74:11)
      at getServerEntry (/Users/dew/remix-vite-monorepo/remix-vite-monorepo/node_modules/@remix-run/dev/dist/vite/plugin.js:197:51)
      at async Context.load (/Users/dew/remix-vite-monorepo/remix-vite-monorepo/node_modules/@remix-run/dev/dist/vite/plugin.js:434:20)
      at async Object.load (file:///Users/dew/remix-vite-monorepo/remix-vite-monorepo/node_modules/vite/dist/node/chunks/dep-bb8a8339.js:44323:32)
      at async loadAndTransform (file:///Users/dew/remix-vite-monorepo/remix-vite-monorepo/node_modules/vite/dist/node/chunks/dep-bb8a8339.js:54954:24)
      at async instantiateModule (file:///Users/dew/remix-vite-monorepo/remix-vite-monorepo/node_modules/vite/dist/node/chunks/dep-bb8a8339.js:55951:10)
```

**Comment by hi-ogawa** [comment]
Thanks for the reproduction.
I think this is a duplicate of https://github.com/remix-run/remix/issues/7722. Pnpm workspace also had the same issue but it will be fixed in https://github.com/remix-run/remix/pull/7913
Until the fix is released, you can workaround it by using `remix reveal` to copy the default entry files to to your `app` folder (see https://remix.run/docs/en/main/file-conventions/entry.client)

**Comment by lithdew** [comment]
> Thanks for the reproduction. I think this is a duplicate of #7722. Pnpm workspace also had the same issue but it will be fixed in #7913 Until the fix is released, you can workaround it by using `remix reveal` to copy the default entry files to to your `app` folder (see https://remix.run/docs/en/main/file-conventions/entry.client)

This worked, many thanks.
